src/Symfony/Component/Security/Http/AccessToken/Oidc/OidcTokenHandler.php

```diff
@@ -92,44 +99,51 @@ public function getUserBadgeFrom(string $accessToken): UserBadge
-                }), true, 512, \JSON_THROW_ON_ERROR);
-            } catch (\Throwable $e) {
-                $this->logger?->error('An error occurred while requesting OIDC configuration.', [
-                    'error' => $e->getMessage(),
-                    'trace' => $e->getTraceAsString(),
-                ]);
-
-                throw new BadCredentialsException('Invalid credentials.', $e->getCode(), $e);
-            }
-
-            try {
-                $jwkset = JWKSet::createFromJson(
-                    $this->discoveryCache->get($this->oidcJWKSetCacheKey, function () use ($oidcConfiguration): string {
-                        $response = $this->discoveryClient->request('GET', $oidcConfiguration['jwks_uri']);
+        if ($this->discovery) {
+            $jwkset = new JWKSet([]);
+            foreach ($this->discovery as ['cache' => $discoveryCache, 'client' => $discoveryClient, 'oidcConfigurationCacheKey' => $oidcConfigurationCacheKey, 'oidcJWKSetCacheKey' => $oidcJWKSetCacheKey]) {
+                try {
+                    $oidcConfiguration = json_decode($discoveryCache->get($oidcConfigurationCacheKey, function () use ($discoveryClient): string {
+                        $response = $discoveryClient->request('GET', '.well-known/openid-configuration');
+
+                        return $response->getContent();
+                    }), true, 512, \JSON_THROW_ON_ERROR);
+                } catch (\Throwable $e) {
+                    $this->logger?->error('An error occurred while requesting OIDC configuration.', [
+                        'error' => $e->getMessage(),
+                        'trace' => $e->getTraceAsString(),
+                    ]);
+
+                    throw new BadCredentialsException('Invalid credentials.', $e->getCode(), $e);
+                }
+
+                try {
+                    $json = $discoveryCache->get($oidcJWKSetCacheKey, function () use (
+                        $discoveryClient,
+                        $oidcConfiguration): string {
+                        $response = $discoveryClient->request('GET', $oidcConfiguration['jwks_uri']);
                         // we only need signature key
                         $keys = array_filter($response->toArray()['keys'], static fn (array $key) => 'sig' === $key['use']);
 
                         return json_encode(['keys' => $keys]);
-                    })
-                );
-            } catch (\Throwable $e) {
-                $this->logger?->error('An error occurred while requesting OIDC certs.', [
-                    'error' => $e->getMessage(),
-                    'trace' => $e->getTraceAsString(),
-                ]);
-
-                throw new BadCredentialsException('Invalid credentials.', $e->getCode(), $e);
+                    });
+
+                    foreach (JWKSet::createFromJson($json) as $key) {
+                        $jwkset = $jwkset->with($key);
+                    }
+                } catch (\Throwable $e) {
```

Comment:
let's change this to that (same above)
unless there's a known runtime reason to catch throwable, we shouldn't by default

```suggestion
                } catch (\Exception $e) {
```